specify/specify7
# Add new feature to change taxon tree type in tree viewer

Fixes #4829
Fixes #4978 
Fixes #4979 

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions
==> you need to add a new taxon tree in your db to test this PR 

Tree Viewer
- Open the taxon tree viewer 
- [ ] If the db as only one taxon tree, verify that the header is a H2 
- [ ] If the db as several taxon trees, verify that the header is a drop down
- [ ] Verify that the header drop down contain all the trees of the db 
- Change the tree type
- [ ] Verify the tree has changed 
- click on a node, add a child 
- [ ] verify that the ranks in the drop down are from the correct tree 

QB
- Open the a new Taxon query 
- [ ] Verify it is the same case when doing: New CO query > Determination > Taxon 
- [ ] Query on any rank from any tree
- [ ] Query on a rank present in 2 or more trees and verify all nodes are doing return
